File: src/web/magmac/app/compile/error/error/CompileErrors.ts
import { Err } from "../../../../../magmac/api/result/Err";
import { CompileResult } from "../../../../../magmac/app/compile/error/CompileResult";
import { CompileResults } from "../../../../../magmac/app/compile/error/CompileResults";
import { NodeContext } from "../../../../../magmac/app/compile/error/context/NodeContext";
import { StringContext } from "../../../../../magmac/app/compile/error/context/StringContext";
import { Node } from "../../../../../magmac/app/compile/node/Node";
import { ImmutableCompileError } from "../../../../../magmac/app/error/ImmutableCompileError";
export class CompileErrors {
	createNodeError(message : String, node : Node) : CompileResult<T> {return 0.fromResult( new Err<>( new ImmutableCompileError( 0, new NodeContext( 0))));;}
	createStringError(message : String, context : String) : CompileResult<T> {return 0.fromResult( new Err<>( new ImmutableCompileError( 0, new StringContext( 0))));;}
}

File: packages/compiler-java/src/main/java/magmac/app/compile/error/type/CompileErrors.java
package magmac.app.compile.error.type;

import magmac.api.result.Err;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.error.CompileResults;
import magmac.app.compile.error.context.NodeContext;
import magmac.app.compile.error.context.StringContext;
import magmac.app.compile.node.Node;
import magmac.app.error.ImmutableCompileError;

public final class CompileErrors {
    public static <T> CompileResult<T> createNodeError(String message, Node node) {
        return CompileResults.fromResult(new Err<>(new ImmutableCompileError(message, new NodeContext(node))));
    }

    public static <T> CompileResult<T> createStringError(String message, String context) {
        return CompileResults.fromResult(new Err<>(new ImmutableCompileError(message, new StringContext(context))));
    }
}
